src/Common/DenseHashMap.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+#include <unordered_map>
+
+#if defined(ARCADIA_BUILD)
```

Comment:
This is extraordinarily ugly.

Need to add comments with a short essay about "What is Arcadia and why do we need this uglyness".

Reply:
Other code LGTM.

Reply:
> This is extraordinarily ugly.

Agree